### PR TITLE
test(bigtable): switch to cmp for comparison

### PR DESCRIPTION
**Pull request by kolea2**

### CODE REVIEW

**Comment by crwilcox** [comment]
For context my addition of EncryptionInfo, #3899, added a pointer to EncryptionInfo, which caused tests that compare values returned from listbackups and getbackup directly. As the pointers are different, that failed. This method will allow our tests to compare values, not pointer addr.

**Review by crwilcox — APPROVED** [review]

**Review by tritone — APPROVED** [review]

File: bigtable/admin.go [hunk 1]
```diff
@@ -1769,6 +1769,35 @@ func (ac *AdminClient) BackupInfo(ctx context.Context, cluster, backup string) (
 	return newBackupInfo(resp)
 }
 
+func (info *BackupInfo) Equal(info2 *BackupInfo) bool {
```
Comment (codyoss) [inline bigtable/admin.go:1772]: This seems very Java-like to me. We don't typically have Equals methods in Go. Can we make use of `cmp` or add a testing helper instead?

**Review by codyoss — CHANGES_REQUESTED** [review]

File: bigtable/admin.go (hunk 1 above)
Comment (kolea2) [reply bigtable/admin.go]: that worked thanks! Updated, PTAL @codyoss

**Review by codyoss — APPROVED** [review]

**Review by crwilcox — APPROVED** [review]

**Review by codyoss — APPROVED** [review]
LGTM thanks!

**Pull request merged**